src/nxdoc/nuxeo-server/upgrading-the-nuxeo-platform/how-to-upgrade-from-lts-2023-to-lts-2025/how-to-upgrade-to-jakarta-rs-3.md

```diff
@@ -0,0 +1,257 @@
+title: How to upgrade to Jakarta RS 3 - WebEngine
+description: Instruction to upgrade your Nuxeo Project to LTS 2025 - Jakarta RS part
+review:
+   comment: ''
+   date: '2025-03-12'
+   status: ok
+labels:
+    - release-notes
+toc: true
+tree_item_index: 200
+---
+
+## Introduction
+
+{{! excerpt}}
+While the upgrade to **Jakarta RS 3** is part of the Jakarta EE 10 one, it has its own section because the implementation
+(Jersey), whose Nuxeo WebEngine is leveraging, has more changes.
+
+Nuxeo WebEngine itself has received breaking changes to benefit from the new Jakarta RS additions.
+{{! /excerpt}}
+
+You must first follow the [How to upgrade to Jakarta EE 10]({{page page='how-to-upgrade-to-jakarta-ee-10'}}) page.
+
+_JAX-RS_ occurrences have been removed from the Nuxeo Server code, it has been renamed to _REST_ where it makes sense.
+
+## Maven Dependency
+
+In addition to the Jakarta RS Specification and Implementation dependencies to change, you will need to update:
+- `nuxeo-webengine-jaxrs` to `nuxeo-webengine-rest`
+- `nuxeo-importer-jaxrs` to `nuxeo-importer-rest`
+- `nuxeo-target-platforms-jaxrs` to `nuxeo-target-platforms-rest`
+- `nuxeo-template-rendering-jaxrs` to `nuxeo-template-rendering-rest`
+
+## Java Code
+
+One of the biggest change in the Jersey Implementation is the total rework of their Dependency Injection engine.
+It is now more extendable, and better fits with the direction of Jakarta RS specification to support `jakarta.inject-api`
+annotations, such as `@Singleton` which is currently used.
+
+### Package
+
+The following Nuxeo packages have changed in order to get rid of _JAX-RS_ occurrences:
+- `org.nuxeo.ecm.webengine.jaxrs` to `org.nuxeo.ecm.webengine.rest`
+- `org.nuxeo.ecm.restapi.server.jaxrs` to `org.nuxeo.ecm.restapi.server`
+- `org.nuxeo.ecm.automation.jaxrs.io` to `org.nuxeo.ecm.automation.io.rest`
+- `org.nuxeo.ecm.automation.server.jaxrs` to `org.nuxeo.ecm.automation.server.rest`
+- `org.nuxeo.targetplatforms.jaxrs` to `org.nuxeo.targetplatforms.rest`
+- `org.nuxeo.template.jaxrs` to `org.nuxeo.template.rest`
+
+In the case you were extending the Nuxeo REST API, you must also update the package `org.nuxeo.ecm.restapi.server.jaxrs` 
+in your project to `org.nuxeo.ecm.restapi.server` to make the Package Scan work correctly.
+
+### Nuxeo Component
+
+The following Nuxeo bundle names have changed in order to get rid of _JAX-RS_ occurrences:
+- `org.nuxeo.ecm.webengine.jaxrs` to `org.nuxeo.ecm.webengine.rest`
+- `org.nuxeo.targetplatforms.jaxrs` to `org.nuxeo.targetplatforms.rest`
+- `org.nuxeo.template.manager.jaxrs` to `org.nuxeo.template.manager.rest`
+
+You must update their occurrences in `@Deploy` annotations in tests.
+
+### Message Body Writer Selection Algorithm
+
+The way the `MessageBodyWriter` is chosen has changed in this upgrade, you can read more in the [Jakarta Documentation](https://eclipse-ee4j.github.io/jersey.github.io/documentation/latest/message-body-workers.html#providers-selection).
```

Comment:
do you mean "is retrieved" or "is selected"?

Reply:
I would have say both, it is selected then retrieved to marshall the response.